piece.gd

```diff
@@ -0,0 +1,133 @@
+class_name Piece
+
+extends Node2D
+
+@export var block_scene: PackedScene
+
+## Represents piece type as name of piece
+var type: int
+
+## Holds block arrangements as JSON
+var block_type_data
+
+## Loads block colors
+var block_textures = [
+	preload("res://assets/Yellow.png"),
+	preload("res://assets/Purple.png"),
+	preload("res://assets/Orange.png"),
+	preload("res://assets/Blue.png"),
+	preload("res://assets/Green.png"),
+	preload("res://assets/Red.png"),
+	preload("res://assets/Cyan.png")
+]
+
+
+## Initialize piece.
+func _ready():
+	block_type_data = [
+		{
+			"name": "square",
+			"variable_block_positions":
+			[$variable_block_position_1.position, $variable_block_position_2.position],
+			"rotation_center": $rotation_center_1,
+			"initial_offset": Vector2(-16, -32)
```

Comment:
IMO, this coordinate system will be problematic when interacting with the Tilemap, which will be a 2D Matrix with coordinates [0:10][0:20]. Either we'll need to add a function to map between pixels <-> Tilemap coordinates or see if the Node2D scene provides a way to convert the coordinates to a Tile coordinate.